client/src/components/__tests__/LoginModal.test.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import LoginModal from '../LoginModal';
+
+import { testInputFieldsIn } from './testUtils';
+
+describe('<LoginModal /> form', () => {
+  const toggleModal = jest.fn();
+  const wrapper = mount(<LoginModal show toggleModal={toggleModal} />);
+
+  test('should contain <form /> element', () => {
+    expect(wrapper.exists('form')).toBe(true);
+  });
+
+  test('form should have onsubmit method', () => {
+    expect(wrapper.find('form').props().onSubmit).toBeInstanceOf(Function);
+  });
+});
+
+describe('input fields in form', () => {
+  const toggleModal = jest.fn();
+  const wrapper = mount(<LoginModal show toggleModal={toggleModal} />);
+  const inputFields = wrapper.find('input');
+  const labels = wrapper.find('label');
+
+  test('should have 3 input fields and labels', () => {
+    expect(inputFields.length).toBe(3);
+    expect(labels.length).toBe(3);
+  });
+
+  test('should autofocus first input field', () => {
+    expect(inputFields.at(0).props().autoFocus).toBe(true);
+  });
+
+  test('should have email input field with onChange and label', () => {
+    const expectedName = 'email';
+    const expectedType = 'email';
+    const expectedLabel = 'Email';
+    testInputFieldsIn(wrapper, 0, { expectedName, expectedType, expectedLabel });
+  });
+
+  test('should have password input field with onChange', () => {
+    const expectedName = 'password';
+    const expectedType = 'password';
+    const expectedLabel = 'Password';
+    testInputFieldsIn(wrapper, 1, { expectedName, expectedType, expectedLabel });
+  });
+  test('should have remember me checkbox field with onChange', () => {
+    const expectedName = 'remember-me';
+    const expectedType = 'checkbox';
+    const expectedLabel = 'Remember Me';
+    testInputFieldsIn(wrapper, 2, { expectedName, expectedType, expectedLabel });
+  });
+});
+
+describe('<LoginModal /> contains cancel and submit button', () => {
+  const toggleModal = jest.fn();
+  const wrapper = mount(<LoginModal show toggleModal={toggleModal} />);
+  const buttons = wrapper.find('button');
+
+  test('should contain two buttons', () => {
+    expect(buttons.length).toBe(2);
+  });
+
+  test('should contain cancel and submit button', () => {
+    const cancelBtn = buttons.at(0);
+    expect(cancelBtn.text().toLowerCase()).toContain('cancel');
```

Comment:
Yes, for good measure. Should I remove it?

Reply:
Actually, I kind of agree that these tests are redundant. But I'm still debating with myself whether they're needed or not. Since, LoginModal uses FormDialog, which is already well tests, we don't  need to test the exact same thing here. Or do we?

Reply:
Yeah you don't need to test the already tested code.

Reply:
Fixed this. Now, I'm just checking for cancel and submit in  FormDialog..

And in the SignupModal and LoginModal, I just check if FormDialog exists, so we know that all things that are in FormDialog are also there in the these modals.